rapid7_insightidr/komand_rapid7_insightidr/actions/query/action.py

```diff
@@ -0,0 +1,57 @@
+import komand
+from .schema import QueryInput, QueryOutput, Input, Output, Component
+# Custom imports below
+from komand.exceptions import PluginException
+from komand_rapid7_insightidr.util.endpoints import QueryLogs
+from komand_rapid7_insightidr.util.resource_helper import ResourceHelper
+import json
+import time
+
+
+class Query(komand.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name='query',
+            description=Component.DESCRIPTION,
+            input=QueryInput(),
+            output=QueryOutput())
+
+    def run(self, params={}):
+        time_now = int(time.time())
+        request = ResourceHelper(self.connection.session, self.logger)
+        request_params = {
+            "from": (time_now - 7776000) * 1000,
+            "to": time_now * 1000
+        }
+        response = request.resource_request(
+            QueryLogs.get_query_logs(self.connection.url, params.get(Input.ID)),
+            'get',
+            params=request_params
+        )
+
+        try:
+            result = json.loads(response['resource'])
+            if response["status"] == 202:
+                response = request.resource_request(result["links"][0]["href"], 'get', params=request_params)
+                result = json.loads(response['resource'])
+        except json.decoder.JSONDecodeError:
```

Comment:
```suggestion
        except (json.decoder.JSONDecodeError, IndexError, KeyError):
```
Need to try/except on `IndexError` and `KeyError` here for the instance that line 36 has a reference error.

Reply:
If status is 202 API return this keys. If there is an error in API there will not be status 202. But I can add this.